gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/dsl/JvmApplicationDistributions.kt

```diff
@@ -32,4 +32,9 @@ abstract class JvmApplicationDistributions : AbstractDistributions() {
     fun windows(fn: Action<WindowsPlatformSettings>) {
         fn.execute(windows)
     }
+    
+    internal val fileAssociations: MutableSet<FileAssociation> = mutableSetOf()
+    fun fileAssociation(mimeType: String, extension: String, description: String) {
```

Comment:
Let's move it to https://github.com/JetBrains/compose-multiplatform/blob/67da8d12f9b18f6aefe68a449cb2ed8c3eb8646f/gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/dsl/PlatformSettings.kt#L19

Other common properties are there, and we can specify them separately.

Reply:
But is it common? It works generally only on desktop, which is JVM. Or should it also work on MacOS Native?

Reply:
I meant common for desktop platforms. `PlatformSettings` is only for desktop.

Native distribution doesn't support MacOs Native, but if there will be one, it will read `PlatformSettings`

Reply:
I have never tried creating MacOS Kotlin/Native. Does it use user Info.plist as the iOS target does, or a generated one as Desktop/JVM one does for MacOS? If the first option takes place, we cannot add support for MacOS native for the feature.

Reply:
But then I'll have to specify the associations for each platform separately.
<img width="1265" alt="image" src="https://github.com/JetBrains/compose-multiplatform/assets/55230817/23083cd6-a6e9-48ab-9208-d3c026f0a4b7">

Reply:
Yes. This was a limitation before for other properties as well (packageVersion, for example). We shouldn't fix this limitation in this feature, otherwise we'll have inconsistent API.

If you need to specify it for all platforms, you can extract an extension function `AbstractPlatformSettings.appFileAssociations()`

Reply:
But icon format is different for each window type. Maybe, I can add an extension which will add the default option for each platform?

Reply:
This is an issue, agree. We have the same issue with the application icon, but it is less painful, as it is a single icon.

A simple extension won't work, as it is called on the configuration time, not on the task execution time (the host OS can be different from the OS for which we build in the future).

We have to provide some `Provider<>` and set an appropriate extension when  `package*` task is called. Not sure how it should look like - we need to try.

Alternatively we can:
1. Delegate this to the user:
```kotlin
fun AbstractPlatformSettings.appFileAssociations(iconExtension) {
   fileAssociation(..., iconFile = project.file("icons/txt.$iconExtension"))
}

windows.appFileAssociations(iconExtension = "ico")
macOs.appFileAssociations(iconExtension = "icns")
linux.appFileAssociations(iconExtension = "png")
```

2. Add extension automatically if it isn't set:
```kotlin
fun AbstractPlatformSettings.appFileAssociations(iconExtension) {
   fileAssociation(..., iconFile = project.file("icons/txt"))
}
```

I would choose 2 as the simplest solution to implement and use. With this approach, I also don't mind to have it on `nativeDistributions` level (I looked at the code more closely, and it seems both levels are fine, with their own cons, which should be solved not in this feature).

@terrakok, @zhelenskiy, what do you think?